java/daikon/FileIO.java

```diff
@@ -1179,9 +1180,21 @@ private static void warn_if_hierarchy_mismatch(PptMap all_ppts) {
       try {
         return chicSocket.getInputStream();
       } catch (IOException e) {
+        try {
+          chicSocket.close();
+        } catch (IOException closeException) {
+          // do nothing
+        }
         throw new RuntimeException("Unable to get Chicory's input stream", e);
       }
     } catch (IOException e) {
```

Comment:
You may need to broaden this to `Exception`. With this change I see
```
../daikon/java/daikon/FileIO.java:1175: error: [required.method.not.called] @MustCall method close may not have been invoked on chicSocket or any of its aliases.
        chicSocket = daikonServer.accept();
                   ^
  The type of object is: java.net.Socket.
  Reason for going out of scope: possible exceptional exit due to daikonServer.close() with exception type java.lang.Exception
```

Reply:
Thanks!  I made this change.

Reply:
@mernst I think the one on this line (1190) needs to change. It looks like you only changed the one on line 1194 below.